openeo/rest/datacube.py

```diff
@@ -1469,17 +1469,20 @@ def save_result(self, format: str = "GTiff", options: dict = None) -> 'DataCube'
             }
         )
 
-    def download(self, outputfile: Union[str, pathlib.Path, None] = None, format: str = "GTIFF", options: dict = None):
+    def download(self, outputfile: Union[str, pathlib.Path, None] = None, format: str = None, options: dict = None):
         """
         Download image collection, e.g. as GeoTIFF.
         If outputfile is provided, the result is stored on disk locally, otherwise, a bytes object is returned.
         The bytes object can be passed on to a suitable decoder for decoding.
 
         :param outputfile: Optional, an output file if the result needs to be stored on disk.
-        :param format: Optional, defaults to "GTIFF", an output format supported by the backend.
+        :param format: Optional, an output format supported by the backend.
         :param options: Optional, file format options
         :return: None if the result is stored to disk, or a bytes object returned by the backend.
         """
+        if format == None:
+            format = guess_format(outputfile = outputfile)
```

Comment:
some python tips/nitpicking:
- checking for `None`  should be done with `x is None` instead of `x == None` because `None` is an object and not a value (roughly speaking)
- a common pattern to simplify this kind of default handling `if` construct is to write it as `format = format or guess_format(outputfile)`